### PR TITLE
Update documentation to more accurately reflect what slide is

### PR DESCRIPTION
**Pull request by ayazhafiz**
Going to release 0.0.2 relatively soon (just have a few issues to
close), before then I want to update the documentation to consolidate
simsimplification/compilation/optimization/lowering terms and express
slide as what it really is, namely a rewrite engine.

ptal @lukebhan

### CODE REVIEW

File: libslide/src/lib.rs [hunk 1]
```diff
@@ -75,8 +76,11 @@
 //! Because of these limitations in string rules, slide also supports function rules. These are
 //! functions of the form
 //!
-//! ```ignore
+//! ```
+//! # use libslide::Expr;
+//! # type _A =
 //! fn(expr: Expr) -> Option<Expr>
+//! # ;
 //! ```
```
Comment (ayazhafiz) [inline libslide/src/lib.rs:83]: doing this so that rustdoc accepts it as rust code and does not display it as "ignored". #s are added to tell rustdoc to ignore displaying those lines

**Comment by ayazhafiz** [comment]
@timothycobrien idk if youre still following this but your reviews are also appreciated if you are willing

**Comment by timothycobrien** [comment]
> @timothycobrien idk if youre still following this but your reviews are also appreciated if you are willing

I am @ayazhafiz just been swamped lately, hope to get back more in the swing of things in the coming weeks

File: README.md [hunk 2]
```diff
@@ -8,16 +8,24 @@
...
 
-slide should be able emit the lowered expression `x`.
+slide should be able emit the simplified expression `x`.
 
-One of slide's design goals is compilation as a platform, where optimizations are configurable
-plugins.
+slide's design goals include
+
+- Simplification as a platform, where rewrite rules are user-configurable.
+    For example, you should be able to give to (or remove from!) slide a rule like `x^2 dx -> 2x`,
```
Comment (lukebhan) [inline README.md:22]: NIT: Don't use you should be able to, I would rephrase as
```
For example, users are able to give to (or remove from!) slide a rule like ... 
```

**Review by lukebhan — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ayazhafiz) [reply README.md:22]: well they are not able to yet ):
